appserver/extras/docker-images/micro/src/main/docker/Dockerfile

```diff
@@ -4,23 +4,29 @@ FROM @docker.java.image@
 EXPOSE 6900 8080
 
 # Configure environment variables
-ENV PAYARA_HOME=/opt/payara\
-    DEPLOY_DIR=/opt/payara/deployments
+# PAYARA_HOME is deprecated - it is here for backward compatibility
+ENV PAYARA_HOME=/opt/payara
+ENV HOME_DIR=${PAYARA_HOME}
+ENV PAYARA_DIR=${HOME_DIR} \
+    DEPLOY_DIR=/opt/payara/deployments \
+    JVM_ARGS="" \
+    MEM_MAX_RAM_PERCENTAGE=70.0 \
+    MEM_XSS=512k
 
 # Create and set the Payara user and working directory owned by the new user
 RUN true \
-    && addgroup payara \
-    && adduser -D -h ${PAYARA_HOME} -H -s /bin/bash payara -G payara \
+    && addgroup --gid 1000 payara \
+    && adduser --system --uid 1000 --no-create-home --shell /bin/bash --home "${HOME_DIR}" --gecos "" --ingroup payara payara \
     && echo payara:payara | chpasswd \
     && mkdir -p ${DEPLOY_DIR}  \
-    && chown -R payara:payara ${PAYARA_HOME} \
+    && chown -R payara:payara ${HOME_DIR} \
     && true
 
 USER payara
-WORKDIR ${PAYARA_HOME}
+WORKDIR ${HOME_DIR}
 
 COPY maven/artifacts/payara-micro.jar .
 
 # Default command to run
-ENTRYPOINT ["java", "-XX:+UseContainerSupport", "-XX:MaxRAMPercentage=90.0", "-jar", "payara-micro.jar"]
+ENTRYPOINT ["sh", "-c", "java -XX:MaxRAMPercentage=${MEM_MAX_RAM_PERCENTAGE} -Xss${MEM_XSS} -XX:+UseContainerSupport ${JVM_ARGS} -jar payara-micro.jar"]
```

Comment:
Just my 2 cents: I added `tini` in the full server image because any shell is not a proper init daemon. You might argue that Docker can use a tini-wrapper, but IMHO it should be done within the image, as there are more container and orchestration environments, which don't wrap it.

Reply:
We use signed tini in the server-full and I have to admit I don't like it from two reasons:

1. gpg used to sign tini crashes builds from time to time and makes them much slower.
2. yesterday I found that it ignores system signals, I'm not sure if it is caused by tini usage or not. But I think we should target these things in standalone issue and stop extending this, because this was originally targeting only better usability and compatibility.

The shell here is only because environment options are not resolved without it.

Generally I agree that all docker files should use same principles only customized to their own usage.